src/ar_firewall.erl

```diff
@@ -1,71 +1,144 @@
 -module(ar_firewall).
--export([start/0, scan_tx/2]).
+-export([start/0, reload/0, scan_tx/1]).
 -include("ar.hrl").
 -include("av/av_recs.hrl").
 -include_lib("eunit/include/eunit.hrl").
 
 %%% Arweave firewall implementation.
 
-%% State definition.
-%% Should store compiled binary scan objects.
+%% Stores compiled binary scan objects.
 -record(state, {
-	sigs = [] % the set of known signatures to filter against
+	tx_id_sigs = [], % a set of known signatures to filter transaction identifiers against
+	content_sigs = [] % a set of known signatures to filter transaction content against
 }).
 
 %% @doc Start a firewall node.
 start() ->
-	spawn(
-		fun() ->
-			server(#state {
-				sigs = av_sigs:all()
-			})
-		end
-	).
+	Firewall = whereis(default_firewall),
+	case Firewall of
+		undefined ->
+			F = spawn(
+				fun() ->
+					server(#state {
+						tx_id_sigs = av_sigs:load(ar_meta_db:get(transaction_blacklist)),
+						content_sigs = av_sigs:load(ar_meta_db:get(content_policies))
+					})
+				end
+			),
+			erlang:register(default_firewall, F);
+		_ ->
+			do_nothing
+	end.
+
+reload() ->
+	Firewall = whereis(default_firewall),
+	case Firewall of
+		undefined ->
+			do_nothing;
+		_ ->
+			erlang:unregister(default_firewall),
```

Comment:
There is no need to create this gap when no firewall is running. We can instead start the new firewall first, register it and then stop the old one.

Reply:
✅

Reply:
Oh turns out it's not possible to atomically reregister a name, so we still have a race condition here. It's probably good enough but we can easily get rid of the race condition by keeping the old process and just feeding it with a new state. I'm ok leaving it like this is how we do it in `ar_http_iface_server:reregister/2`.

Reply:
Yeah, there is still a tiny gap, I think it's fine.

Reply:
✅